src/main/java/bgu/spl/net/impl/stomp/StompEncDec.java

```diff
@@ -0,0 +1,82 @@
+package bgu.spl.net.impl.stomp;
+
+import bgu.spl.net.api.MessageEncoderDecoder;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class StompEncDec implements MessageEncoderDecoder<Frame> {
+
+    private byte[] bytes = new byte[1 << 10]; //start with 1k
+    private int len = 0;
+
+    /**
+     *   Converts a String to the Frame object representing it
+     * @param msg The message as a string
+     * @return The Frame representation
+     */
+    private Frame parseMessage(String msg) {
+        int delim = 0, prevDelim = 0;
+        int msgPart = 1;
+        Frame parsed = null;
+        do {
+            prevDelim = delim;
+            delim = msg.indexOf("\n", prevDelim+1); // Find the end of the next line
+            if (msgPart == 1) { // The command
+                parsed = new Frame(msg.substring(0, delim));
+                msgPart++;
+            }
+            else if (msgPart == 2) { // headers
+                int mid = msg.indexOf(":", prevDelim);
+                if(mid == -1) {
+                    msgPart++;
+                    continue;
+                }
+                parsed.addHeader(msg.substring(prevDelim + 1, mid), msg.substring(mid + 1, delim));
+            } // Body and return
+            else if (msgPart == 3) {
+                // -2 so we don't include the \n or the \0
+                parsed.addBody(msg.substring(prevDelim + 1));
+                return parsed;
+            }
+
+
+        } while (delim < msg.length());
+        return null; // Impossible
```

Comment:
maybe return parsed here, and in (msgPart == 3) add delim = msg.length()
or even better, while (msgPart <3) and after while loop add body

Reply:
Done